### PR TITLE
[fix](tvf view)Support Table valued function view for nereids.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Nereids doesn't support view based table value function, because tvf view doesn't contain the proper qualifier (catalog, db and table name). This pr is to support this function.

Also, fix nereids table value function explain output exprs incorrect bug.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalTVFRelation.java [hunk 1]
```diff
@@ -39,16 +39,19 @@
 public class LogicalTVFRelation extends LogicalRelation implements TVFRelation {
 
     private final TableValuedFunction function;
+    private final ImmutableList<String> qualifier;
 
     public LogicalTVFRelation(RelationId id, TableValuedFunction function) {
         super(id, PlanType.LOGICAL_TVF_RELATION);
         this.function = function;
+        qualifier = ImmutableList.of("_table_valued_function_" + function.getName());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalTVFRelation.java:47]: How about using a shorten prefix? eg: _tvf_ ?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 2]
```diff
@@ -669,7 +669,7 @@ public PlanFragment visitPhysicalTVFRelation(PhysicalTVFRelation tvfRelation, Pl
         // set label for explain
         for (Slot slot : slots) {
             String tableColumnName = "_table_valued_function_" + tvfRelation.getFunction().getName()
-                    + "." + slots.get(0).getName();
+                    + "." + slot.getName();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:672]: Add a case to cover this. U can use `explain action` in regression test

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.44 seconds
 stream load tsv:          541 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162081945 Bytes

File: regression-test/suites/external_table_p2/tvf/test_tvf_view_p2.groovy [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        sql """create view tvf_view as select * from hdfs (
+            "uri"="hdfs://${nameNodeHost}:${hdfsPort}:/usr/hive/warehouse/tpch_1000_parquet.db/part/000091_0",
+            "fs.defaultFS"="hdfs://${nameNodeHost}:${hdfsPort}",
+            "hadoop.username" = "hadoop",
+            "format"="parquet");"""
+
+        qt_1 """select count(*) from tvf_view"""
+        qt_2 """select * from tvf_view order by p_partkey limit 10"""
+        qt_3 """select p_partkey from tvf_view order by p_partkey limit 10"""
+        qt_4 """explain select * from tvf_view"""
```
Comment (morningman) [inline regression-test/suites/external_table_p2/tvf/test_tvf_view_p2.groovy:37]: use `explain` action

**Review by morningman — COMMENTED** [review]
lgtm, but plz modify the test cases

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.79 seconds
 stream load tsv:          541 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162218205 Bytes

**Comment by Jibing-Li** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.1 seconds
 stream load tsv:          538 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162009935 Bytes

**Review by zddr — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**